src/render/gif.rs

```diff
@@ -0,0 +1,277 @@
+use std::{
+    cell::RefCell,
+    fs::File,
+    rc::Rc,
+    sync::atomic::{AtomicUsize, Ordering},
+    time::{Duration, Instant},
+};
+
+use apex_hardware::FrameBuffer;
+
+use embedded_graphics::{
+    image::{Image, ImageRaw},
+    pixelcolor::BinaryColor,
+    prelude::Point,
+    Drawable,
+};
+use gif::Frame;
+
+static GIF_MISSING: &[u8] = include_bytes!("./../../assets/gif_missing.gif");
+
+static DISPLAY_HEIGHT: i32 = 40;
+static DISPLAY_WIDTH: i32 = 128;
```

Comment:
Not a huge fan of hardcoding this here since we might support more devices in the future.

Reply:
oh, I thought it was only the apex' screens, it is only the maximum resolution that it will look up for (so as to not "trust" the arguments, passed in the function, in order not to increase dramatically loading times)

Should I remove it?

Reply:
I think it can stay for now. If we ever do add support for more screens we'll figure something out.